src/integer_object.cpp

```diff
@@ -458,23 +458,32 @@ Value IntegerObject::bitwise_xor(Env *env, Integer &self, Value arg) {
 }
 
 Value IntegerObject::left_shift(Env *env, Integer &self, Value arg) {
+    if (is_zero(self))
+        return Value::integer(0);
     auto integer = arg.to_int(env);
     if (is_bignum(integer)) {
-        if (IntegerObject::is_negative(self))
+        if (IntegerObject::is_negative(self) && IntegerObject::is_negative(integer))
```

Comment:
It doesn't matter in the slightest, but in the future, I hope to convert all the `IntegerObject::is_negative(integer)` type calls to the equivalent and easier-on-the-eyes `integer.is_negative()` where possible.